distribution/lib/Standard/Base/0.0.0-dev/src/Data/Hash_Map.enso

```diff
@@ -0,0 +1,117 @@
+import project.Data.Numbers.Integer
+import project.Data.Vector.Vector
+import project.Data.Pair.Pair
+
+from project.Data.Boolean import Boolean, True, False
+from project import Error, Nothing, Any, Panic
+from project.Error.No_Such_Key import No_Such_Key
+
+
+@Builtin_Type
+type Hash_Map key value
```

Comment:
> I strongly suggest to drop Hash_ prefix and have this as "the Enso Map"

Yes, I definitely plan to replace our `Map` with this.

> Nobody knows how JavaScript map is implemented

Right. I will hide the references to hashing as much as possible.

Note that even though nobody knows how JavaScript map is implemented, there is a way for any object to effectively override "hash_code" by overriding `toString` method, because JavaScript map first invokes `toString` method on an object before it uses it as a key. See https://stackoverflow.com/questions/194846/is-there-hash-code-function-accepting-any-object-type .